components/transactions/TransactionDetails.js

```diff
@@ -74,11 +81,24 @@ const TransactionDetails = ({ displayActions, transaction, isRoot, isHostAdmin,
     order,
     expense,
     isRejected,
+    createdAt,
   } = transaction;
-  const userCanRejectOrRefund = isRoot || isHostAdmin;
+  // calculate 30 days ago from transaction
+  const now = Date.now();
+  const thirtyDaysInMilliseconds = new Date(1000 * 60 * 60 * 24 * 30);
+  const timeLimitInMilliseconds = new Date(now - thirtyDaysInMilliseconds);
+  const createdAtInMilliseconds = Date.parse(createdAt);
+  const transactionIsOlderThanThirtyDays = createdAtInMilliseconds < timeLimitInMilliseconds;
```

Comment:
This might work but this doesn't look like correct code. In your code:

- `const thirtyDaysInMilliseconds` should be a Number, not a Date. `1000 * 60 * 60 * 24 * 30`
- `const timeLimitInMilliseconds` should also be a Number, not a Date.

I'm generally not complaining about too many variables but in this case I think that a a shorter version will be much clearer:

```
const thirtyDaysInMillisecond = 1000 * 60 * 60 * 24 * 30;

const isTransactionIsOlderThanThirtyDays = Date.parse(createdAt) < (Date.now() - thirtyDaysInMillisecond);
```